controllers/provider-packet/pkg/webhook/controlplane/ensurer.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"context"
+	"github.com/gardener/gardener-extensions/controllers/provider-packet/pkg/packet"
+	"github.com/gardener/gardener-extensions/pkg/webhook/controlplane"
+	"github.com/gardener/gardener-extensions/pkg/webhook/controlplane/genericmutator"
+
+	"github.com/coreos/go-systemd/unit"
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+)
+
+// NewEnsurer creates a new controlplane ensurer.
+func NewEnsurer(logger logr.Logger) genericmutator.Ensurer {
+	return &ensurer{
+		logger: logger.WithName("packet-controlplane-ensurer"),
+	}
+}
+
+type ensurer struct {
+	genericmutator.NoopEnsurer
+	logger logr.Logger
+}
+
+// EnsureKubeAPIServerDeployment ensures that the kube-apiserver deployment conforms to the provider requirements.
+func (e *ensurer) EnsureKubeAPIServerDeployment(ctx context.Context, dep *appsv1.Deployment) error {
+	ps := &dep.Spec.Template.Spec
+	if c := controlplane.ContainerWithName(ps.Containers, "kube-apiserver"); c != nil {
+		ensureKubeAPIServerCommandLineArgs(c)
+		ensureEnvVars(c)
```

Comment:
Since you are adding an env var that uses a secret as a source, you also need to inject the checksum for this secret. See how it's done for the other providers.